nav2_rviz_plugins/include/nav2_rviz_plugins/nav2_panel.hpp

```diff
@@ -146,17 +147,34 @@ class InitialThread : public QThread
 public:
   using SystemStatus = nav2_lifecycle_manager::SystemStatus;
 
-  explicit InitialThread(nav2_lifecycle_manager::LifecycleManagerClient & client)
-  : client_(client)
+  explicit InitialThread(
+    nav2_lifecycle_manager::LifecycleManagerClient & client_nav,
+    nav2_lifecycle_manager::LifecycleManagerClient & client_loc)
+  : client_nav_(client_nav), client_loc_(client_loc)
   {}
 
   void run() override
   {
-    SystemStatus status = SystemStatus::TIMEOUT;
-    while (status == SystemStatus::TIMEOUT) {
-      status = client_.is_active(std::chrono::seconds(1));
+    SystemStatus status_nav = SystemStatus::TIMEOUT;
+    SystemStatus status_loc = SystemStatus::TIMEOUT;
+
+    while (status_nav == SystemStatus::TIMEOUT) {
+      if (status_nav == SystemStatus::TIMEOUT) {
+        status_nav = client_nav_.is_active(std::chrono::seconds(1));
+      }
+    }
+
+    // try to communicate twice, might not actually be up if in SLAM mode
+    bool tried_loc_bringup_once = false;
+    while (status_loc == SystemStatus::TIMEOUT) {
+      status_loc = client_loc_.is_active(std::chrono::seconds(1));
+      if (tried_loc_bringup_once) {
+        break;
+      }
+      tried_loc_bringup_once = true;
     }
-    if (status == SystemStatus::ACTIVE) {
+
```

Comment:
So the user has no way of knowing that localization hasn't been started up, only if navigation startup succeeded, right?

Seems like an enhancement is needed to the panel to show status for both

Reply:
There are logging statements, so there should be some knowledge.

Reply:
If the user is in the RVIZ GUI on a remote machine, they won't see a message in the terminal, right?

I won't block this PR based on this but could we get an enhancement to the nav2 Rviz panel to show status for both in a follow on PR?

Reply:
The error is appearing in the RVIZ terminal.

If your argument is that we can't expect someone to see terminal output, then we shouldn't include any at all anywhere. This isn't unique to this situation.

I think that's a reasonable request -- I'll file a ticket.

Reply:
You're right, I don't know why I was thinking it would appear in the server's terminal. I think we're in agreement now.